sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -281,10 +290,29 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
 }
 
 /**
- * Common base class for [[GetMapValue]] and [[ElementAt]].
+ * Common trait for [[GetMapValue]] and [[ElementAt]].
  */
+trait GetMapValueUtil extends BinaryExpression with ImplicitCastInputTypes {
```

Comment:
Does `GetMapValueUtil` need to extend `extends BinaryExpression with ImplicitCastInputTypes`? If `child` and `key` are required, `computeNullabilityFromMap` can accept the `child` and `key` as parameters.

Reply:
It seems `BinaryExpression.nullSafeCodeGen` is used in the other place?
https://github.com/apache/spark/blob/2d2fb34b93a55af7e45191c8974bb55a0ed76e87/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala#L326

Reply:
Thanks. Then, what about `GetArrayItemUtil`? Actually, the reason why I asked is that this PR makes some non-trivial lineages like the followings.
- `def BinaryExpression.left` -> `private val GetArrayItemUtil.child` -> `override def GetArrayItem.left`. 
- `def BinaryExpression.left` -> `private val GetMapValueUtil.child` -> `override def GetMapValue.left`

Reply:
ah, ok. I'll simplify it.